Modules/clinic/posixmodule.c.h

```diff
@@ -3705,6 +3705,50 @@ os_pread(PyObject *module, PyObject *const *args, Py_ssize_t nargs)
 
 #endif /* defined(HAVE_PREAD) */
 
+#if defined(HAVE_PREADV2)
+
+PyDoc_STRVAR(os_preadv2__doc__,
+"preadv2($module, fd, buffers, offset, flags, /)\n"
```

Comment:
API question: `preadv2` only exists because regular `preadv` was missing a `flags` argument, and in C the only way to add an argument is to make a new function. In python we don't have that problem. So an alternative would be to make this `os.preadv(fd, buffers, offset[, flags])`, and have it call either `preadv` or `preadv2` as appropriate. (`preadv2(..., flags=0)` is equivalent to `preadv`.)

I kind of like this better, since (a) exposing `preadv` is probably a good idea, even on systems that don't have `preadv2`, (b) it reduces clutter in `os`. OTOH the same thing happened for `pipe` and `pipe2`, and it looks like we exposed those as two independent functions in `os`, so ¯\\_(ツ)_/¯

Reply:
https://www.gnu.org/software/libc/manual/html_mono/libc.html#index-preadv2 seems to support @njsmith statement.

Reply:
@njsmith Thanks for the comment! CPython only exposes `pread` and `readv` and currently it does not expose `preadv`. Is your sugestion to implement `preadv2` as `preadv` and call the former if flags are provided and the later (`preadv`) in ther case? 

I am asking this because initially I thought you were saying that `preadv` was available and we should reuse the function

Reply:
Right, I'm doing the classic pull request reviewer move of trying to trick you into doing extra work while you're there :-). We don't currently expose `preadv`, but it's obviously a useful thing, so I'm suggesting you add a `preadv` wrapper, and then add `preadv2` support to that. If we add `preadv2` now alone, then we'll probably end up adding `preadv` later, and thus have a messier api in the end.

Reply:
BTW, it's probably simpler to structure the code like (forgive my mixed Python/c pseudocode):

```
#if defined(HAVE_PREADV) || defined (HAVE_PREADV2)
def preadv(fd, offset, buffers, flags=0):
#ifdef HAVE_PREADV2
    return preadv2(...)
#else
    if flags != 0: raise ...
    return preadv(...)
#endif
```

Since `preadv2` has a superset of the functionality of `preadv`, when it's available you can use it for all flags values, including 0.